src/exceptiongroup/_suppress.py

```diff
@@ -1,13 +1,19 @@
 import sys
 from contextlib import AbstractContextManager
 from types import TracebackType
-from typing import Type
+from typing import TYPE_CHECKING, Type
 
 if sys.version_info < (3, 11):
     from ._exceptions import BaseExceptionGroup
 
+if TYPE_CHECKING:
+    # requires python 3.9
+    BaseClass = AbstractContextManager[None]
+else:
+    BaseClass = AbstractContextManager
 
-class suppress(AbstractContextManager[None]):
+
+class suppress(BaseClass):
```

Comment:
Would pyright report an error if we used `AbstractContextManager[None]` together with `from __future__ import annotations`?

Reply:
nope, but it appears to make no difference and still fail py37 & py38

Reply:
Ah of course, inheritance is not a type hint that can delay/avoid execution